src/hooks/useQuizEvents/useQuizNextClick.ts

```diff
@@ -2,26 +2,39 @@ import { useCallback } from 'react';
 import { ActionAnswerQuestion, QuestionTypes } from '../../components/CioQuiz/actions';
 import { QuizAPIReducerState } from '../../components/CioQuiz/quizApiReducer';
 import { QuizLocalReducerState } from '../../components/CioQuiz/quizLocalReducer';
-import { QuizEventsReturn } from '../../types';
+import { QuizEventsReturn, OnQuizNextQuestion } from '../../types';
+import { isFunction } from '../../utils';
 
 const useQuizNextClick = (
   quizApiState: QuizAPIReducerState,
   quizLocalState: QuizLocalReducerState,
-  dispatchLocalState: React.Dispatch<ActionAnswerQuestion>
+  dispatchLocalState: React.Dispatch<ActionAnswerQuestion>,
+  onQuizNextQuestion?: OnQuizNextQuestion
 ): QuizEventsReturn.NextQuestion => {
   const quizNexClickHandler = useCallback(() => {
     const currentQuestion = quizApiState.quizCurrentQuestion?.next_question;
     const currentQuestionId = currentQuestion?.id;
+
     if (dispatchLocalState && currentQuestionId) {
       const currentAnswerInput = quizLocalState.answerInputs[currentQuestionId];
+
       if (currentAnswerInput?.value?.length || currentQuestion?.type === 'cover') {
         dispatchLocalState({
           type: QuestionTypes.Next,
           payload: quizApiState.quizCurrentQuestion,
         });
       }
+
+      if (currentQuestion && onQuizNextQuestion && isFunction(onQuizNextQuestion)) {
```

Comment:
This is a nitpick. but isFunction already checks onQuizNextQuestion so we don't need to check && onQuizNextQuestion here again 😅

Reply:
All good if typescript yells. I think we can add ! on the quiz function when you call it like onQuizNextQuestion!(...)

But both are fine.

Reply:
You are a wizard @stanlp1 🧙